packages/swayze/lib/src/widgets/headers/gestures/header_gesture_detector.dart

```diff
@@ -250,67 +274,253 @@ class _HeaderGestureDetectorState extends State<HeaderGestureDetector> {
-                originOffset: dragOriginOffsetCache!,
-              );
-
-              handleUpdateSelection(headerGestureDetails);
-            };
-            instance.onEnd = (DragEndDetails details) {
-              dragOriginOffsetCache = null;
-              internalScope.controller.scroll.stopAutoScroll(widget.axis);
-            };
-            instance.onCancel = () {
-              dragOriginOffsetCache = null;
-              internalScope.controller.scroll.stopAutoScroll(widget.axis);
-            };
-          },
-        ),
-        TapGestureRecognizer:
-            GestureRecognizerFactoryWithHandlers<TapGestureRecognizer>(
-          () => TapGestureRecognizer(debugOwner: this),
-          (TapGestureRecognizer instance) {
-            instance.onTapDown = (TapDownDetails details) {
-              final headerGestureDetails = _getHeaderGestureDetails(
-                axis: widget.axis,
-                context: context,
-                globalPosition: details.globalPosition,
-              );
-
-              handleStartSelection(headerGestureDetails);
-            };
-          },
-        ),
+    return MouseRegion(
+      cursor: cursor,
+      onHover: (event) {
+        if (!internalScope.config.isHeaderDragAndDropEnabled) {
+          return;
+        }
+
+        final headerGestureDetails = _getHeaderLocalPositionGestureDetails(
+          axis: widget.axis,
+          context: context,
+          localPosition: event.localPosition,
+        );
+        final isSelected = isHeaderSelected(
+          headerGestureDetails.headerPosition,
+          widget.axis,
+        );
+        setCursorState(
+          isSelected ? SystemMouseCursors.grab : MouseCursor.defer,
+        );
       },
-      behavior: HitTestBehavior.translucent,
+      child: RawGestureDetector(
+        gestures: <Type, GestureRecognizerFactory>{
+          PanGestureRecognizer:
+              GestureRecognizerFactoryWithHandlers<PanGestureRecognizer>(
+            () => PanGestureRecognizer(debugOwner: this),
+            (PanGestureRecognizer instance) {
+              instance.onStart = (DragStartDetails details) {
+                final headerGestureDetails = _getHeaderGestureDetails(
+                  axis: widget.axis,
+                  context: context,
+                  globalPosition: details.globalPosition,
+                );
+
+                final selection = hoverSelection(
+                  headerGestureDetails.headerPosition,
+                  widget.axis,
+                );
+
+                if (selection != null &&
+                    internalScope.config.isHeaderDragAndDropEnabled) {
+                  final range = headerSelectionRange(selection);
+                  handleStartDraggingHeader(details, range);
+                } else {
+                  handleStartSelection(headerGestureDetails);
+                }
+
+                dragOriginOffsetCache = headerGestureDetails.localPosition;
```

Comment:
we commonly have these methods inline, but I wonder if it wouldn't be beneficial to extract them to their own methods where we could add some documentation on their behavior.